src/mon/OSDMonitor.cc

```diff
@@ -5897,7 +5897,7 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     }
 
     err = newcrush.adjust_subtree_weightf(g_ceph_context, id, w);
-    if (err < 0)
+    if (err <= 0)
```

Comment:
Better to wait for a commit cycle in the rare case that it is in flight then to add a bunch of code to tell whether it is not in flight.